qa/L0_lifecycle/lifecycle_test.py

```diff
@@ -3405,6 +3405,296 @@ def test_shutdown_with_live_connection(self):
             "exit timeout countdown restart detected",
         )
 
+    def test_add_custom_config(self):
+        models_base = ("savedmodel", "plan")
+        models_shape = ((1, 16), (1, 16))
+        models = list()
+        for m in models_base:
+            models.append(tu.get_model_name(m, np.float32, np.float32, np.float32))
+
+        # Make sure savedmodel and plan are in the status
+        for model_name in models:
+            try:
+                for triton_client in (
+                    httpclient.InferenceServerClient("localhost:8000", verbose=True),
+                    grpcclient.InferenceServerClient("localhost:8001", verbose=True),
+                ):
+                    self.assertTrue(triton_client.is_server_live())
+                    self.assertTrue(triton_client.is_server_ready())
+                    self.assertTrue(triton_client.is_model_ready(model_name, "1"))
+                    self.assertFalse(triton_client.is_model_ready(model_name, "2"))
+                    self.assertTrue(triton_client.is_model_ready(model_name, "3"))
+            except Exception as ex:
+                self.assertTrue(False, "unexpected error {}".format(ex))
+
+        # Run inference on the model on all versions. Only version 1, 3 should work.
+        for model_name, model_shape in zip(models_base, models_shape):
```

Comment:
You can iterate different backend types and shapes but that is not essential for testing this feature.. I would rather just test one model setting for simplicity and group the version testings below to be:
```
# check available version
for version in expected_available_versions:
    ...
for version in expected_unavailable_versions:
    ...
```

Reply:
Actually, I don't know if the inference check is necessary as you already check the model version readiness. With that, it just seems to be an sanity check for me.

Reply:
Removed unnecessary tests